From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martin Schmiedel <Martin.Schmiedel@tq-group.com>
Date: Thu, 27 Aug 2026 10:49:40 +0200
Subject: arm64: dts: freescale: enable LDB termination resistor for mba8mpxl
 LVDS

mba8mpxl_trixie_edge_7.1.10 showed a doubled, offset LVDS image caused
by LVDS_CTRL.HS_EN (drivers/gpu/drm/bridge/fsl-ldb.c) being unset.

Set nxp,enable-termination-resistor to enable HS_EN. Confirmed on
hardware: the doubled/offset image is gone.

mba8mp-ras314 shares the same SoC, driver, panel and overlay
(imx8mp-tqma8mpql-mba8mpxl-lvds-tm070jvhg33.dtbo) but does not show
the symptom, so it intentionally stays unset there.

Signed-off-by: Martin Schmiedel <Martin.Schmiedel@tq-group.com>
---
 arch/arm64/boot/dts/freescale/imx8mp-tqma8mpql-mba8mpxl-lvds-g133han01.dtso   | 1 +
 arch/arm64/boot/dts/freescale/imx8mp-tqma8mpql-mba8mpxl-lvds-tm070jvhg33.dtso | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-tqma8mpql-mba8mpxl-lvds-g133han01.dtso b/arch/arm64/boot/dts/freescale/imx8mp-tqma8mpql-mba8mpxl-lvds-g133han01.dtso
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-tqma8mpql-mba8mpxl-lvds-g133han01.dtso
+++ b/arch/arm64/boot/dts/freescale/imx8mp-tqma8mpql-mba8mpxl-lvds-g133han01.dtso
@@ -52,6 +52,7 @@ &lvds_bridge {
 	assigned-clock-parents = <&clk IMX8MP_VIDEO_PLL1_OUT>;
 	assigned-clock-rates = <0>, <988400000>;
 	status = "okay";
+	nxp,enable-termination-resistor;
 
 	ports {
 		port@1 {
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-tqma8mpql-mba8mpxl-lvds-tm070jvhg33.dtso b/arch/arm64/boot/dts/freescale/imx8mp-tqma8mpql-mba8mpxl-lvds-tm070jvhg33.dtso
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-tqma8mpql-mba8mpxl-lvds-tm070jvhg33.dtso
+++ b/arch/arm64/boot/dts/freescale/imx8mp-tqma8mpql-mba8mpxl-lvds-tm070jvhg33.dtso
@@ -42,6 +42,7 @@ &lcdif2 {
 
 &lvds_bridge {
 	status = "okay";
+	nxp,enable-termination-resistor;
 
 	ports {
 		port@1 {
-- 
Armbian

